### ISSUE TITLE
GA v2.3.0 can't compile on MacOs anymore

### ISSUE THREAD

**Issue by selimanac** [GameAnalytics/GA-SDK-DEFOLD#24]
Version 2.3.0 can't compile on MacOS anymore(build target: MacOS). This is a blocker. 
Edit: Works fine for iOS and Android builds, fails only when target is MacOS.

Defold version: 1.2.162 - Alpha

![Screen Shot 2019-10-03 at 11 54 52](https://user-images.githubusercontent.com/1324635/66113041-aaa0c080-e5d4-11e9-8b5c-e14400713c8a.png)

Error:
```

Undefined symbols for architecture x86_64:
  "____chkstk_darwin", referenced from:
      gameanalytics::GameAnalytics::addErrorEvent(gameanalytics::EGAErrorSeverity, char const*, char const*) in libGameAnalytics.a(GameAnalytics.o)
      std::__1::function<void ()>::function<gameanalytics::GameAnalytics::addErrorEvent(gameanalytics::EGAErrorSeverity, char const*, char const*)::$_19, void>(gameanalytics::GameAnalytics::addErrorEvent(gameanalytics::EGAErrorSeverity, char const*, char const*)::$_19) in libGameAnalytics.a(GameAnalytics.o)
      gameanalytics::errorreporter::GAUncaughtExceptionHandler::signalHandler(int, __siginfo*, void*) in libGameAnalytics.a(GAUncaughtExceptionHandler.o)
      gameanalytics::http::GAHTTPApi::sendSdkErrorEvent(gameanalytics::http::EGASdkErrorType) in libGameAnalytics.a(GAHTTPApi.o)
      std::__1::future<std::__1::__invoke_of<std::__1::decay<gameanalytics::http::GAHTTPApi::sendSdkErrorEvent(gameanalytics::http::EGASdkErrorType)::$_0>::type>::type> std::__1::async<gameanalytics::http::GAHTTPApi::sendSdkErrorEvent(gameanalytics::http::EGASdkErrorType)::$_0>(std::__1::launch, gameanalytics::http::GAHTTPApi::sendSdkErrorEvent(gameanalytics::http::EGASdkErrorType)::$_0&&) in libGameAnalytics.a(GAHTTPApi.o)
      _tinfl_decompress_mem_to_heap in libGameAnalytics.a(miniz.o)
      _tinfl_decompress_mem_to_mem in libGameAnalytics.a(miniz.o)
      ...
  "_objc_opt_respondsToSelector", referenced from:
      getOSXVersion() in libGameAnalytics.a(GADeviceOSX.o)

```

[Full error log](https://github.com/GameAnalytics/GA-SDK-DEFOLD/files/3685214/error.txt)

**Comment by the1schwartz** [comment]
Ok sorry. I will look into this straight away.

**Comment by the1schwartz** [comment]
Seems to be because the C++ was built with XCode 11 and Defold build server doesn't like that apparently. Will continue to investigate for a solution.

**Comment by selimanac** [comment]
I'm not sure if this relevant but they are updating their build server:
https://forum.defold.com/t/extender-server-ios-13-osx-10-15-clang-9/63569

**Comment by the1schwartz** [comment]
Ok thanks. Hmm, yeah it could be it maybe.

**Comment by the1schwartz** [comment]
Until they have made the release you can use the previous version to work with if needed: https://github.com/GameAnalytics/GA-SDK-DEFOLD/archive/2.2.8.zip

**Comment by selimanac** [comment]
Thank you. I'll let them know about this issue.

**Comment by selimanac** [comment]
@the1schwartz I did a quick test. It worked with upcoming version of editor and build server. But until then it is going to fail for users.

**Comment by the1schwartz** [comment]
ok i will add the previous version for now then. thanks for testing

**Comment by the1schwartz** [comment]
This should be fixed now in the latest version now (v2.3.1). Let me know if it still happens for you.
